dotcom-rendering/src/lib/content.d.ts

```diff
@@ -290,6 +290,7 @@ interface NewsletterSignupBlockElement {
 	_type: 'model.dotcomrendering.pageElements.NewsletterSignupBlockElement';
 	newsletter: Newsletter;
 	elementId: string;
+	clearFromLeftCol?: boolean;
```

Comment:
Is there a less imperative way of achieving this? This prop is telling the code what css to set but a more declarative (easier to scale) approach would be to tell the component about the context and allow it to make the decision about how to manage this.

Reply:
Yes, good call - the decision is based on the type of the last floating element before the point the EmailSignUp is being inserted, so I'll pass the component that type ('supporting', 'richlink', 'thumbnail' or undefined) and have the component set its own CSS based on that.